src/System.ComponentModel.TypeConverter/tests/ContainerTests.cs

```diff
@@ -0,0 +1,765 @@
+		{
+			ValidateName (component, name);
+		}
+
+		protected override void ValidateName (IComponent component, string name)
+		{
+			if (AllowDuplicateNames)
+				return;
+			base.ValidateName (component, name);
+		}
+
+		public bool Contains (IComponent component)
+		{
+			bool found = false;
+			
+			foreach (IComponent c in Components) {
+				if (component.Equals (c)) {
+					found = true;
+					break;
+				}
+			}
+			return found;
+		}
+
+		public new void Dispose (bool disposing)
+		{
+			base.Dispose (disposing);
+		}
+	}
+	
+	class TestComponent : Component {
+		public override ISite Site {
+			get {
+				return base.Site;
+			}
+			set {
+				base.Site = value;
+				if (value != null) {
+					Assert.NotNull (value.GetService (typeof (ISite)));
+					Assert.NotNull (value.GetService (typeof (TestService)));
+				}
+			}
+		}
+
+		public bool IsDisposed {
+			get { return disposed; }
+		}
+
+		public bool ThrowOnDispose {
+			get { return throwOnDispose; }
+			set { throwOnDispose = value; }
+		}
+
+		protected override void Dispose (bool disposing)
+		{
+			if (ThrowOnDispose)
+				throw new InvalidOperationException ();
+
+			base.Dispose (disposing);
+			disposed = true;
+		}
+
+		private bool disposed;
+		private bool throwOnDispose;
+	}
+
+	
+	public class ContainerTest
+	{
+		private TestContainer _container;
+
+		public ContainerTest()
+		{
+			_container = new TestContainer ();
+		}
+
+		[Fact] // Add (IComponent)
+		public void Add1 ()
+		{
+			TestContainer containerA = new TestContainer ();
+			TestContainer containerB = new TestContainer ();
+
+			ISite siteA;
+			ISite siteB;
+
+			TestComponent compA = new TestComponent ();
+			Assert.Null (compA.Site);
+			TestComponent compB = new TestComponent ();
+			Assert.Null (compB.Site);
+			Assert.Equal (0, containerA.Components.Count);
+			Assert.Equal (0, containerB.Components.Count);
+
+			containerA.Add (compA);
+			siteA = compA.Site;
+			Assert.NotNull (siteA);
+			Assert.Same (compA, siteA.Component);
+			Assert.Same (containerA, siteA.Container);
+			Assert.False (siteA.DesignMode);
+			Assert.Null (siteA.Name);
+			containerA.Add (compB);
+			siteB = compB.Site;
+			Assert.NotNull (siteB);
+			Assert.Same (compB, siteB.Component);
+			Assert.Same (containerA, siteB.Container);
+			Assert.False (siteB.DesignMode);
+			Assert.Null (siteB.Name);
+
+			Assert.False (object.ReferenceEquals (siteA, siteB));
+			Assert.Equal (2, containerA.Components.Count);
+			Assert.Equal (0, containerB.Components.Count);
+			Assert.Same (compA, containerA.Components [0]);
+			Assert.Same (compB, containerA.Components [1]);
+
+			// check effect of adding component that is already member of
+			// another container
+			containerB.Add (compA);
+			Assert.False (object.ReferenceEquals (siteA, compA.Site));
+			siteA = compA.Site;
+			Assert.NotNull (siteA);
+			Assert.Same (compA, siteA.Component);
+			Assert.Same (containerB, siteA.Container);
+			Assert.False (siteA.DesignMode);
```

Comment:
Nit: do you want to have a message with these Assert.True/Falses as you do elsewhere?